airflow-core/src/airflow/utils/log/file_task_handler.py

```diff
@@ -610,6 +610,11 @@ def _read(
         local_logs: list[RawLogStream] = []
         executor_logs: list[RawLogStream] = []
         served_logs: list[RawLogStream] = []
+        is_in_not_finished_state = ti.state in (
+            TaskInstanceState.RUNNING,
+            TaskInstanceState.DEFERRED,
+            TaskInstanceState.UP_FOR_RETRY
```

Comment:
I'm not convinced by this -- if a task is un UP_FOR_RETRY, it is no longer actively running on a worker, and as far as logs are concerned it should be in exactly the same as if it was in SUCCESS or FAILED state.